src/aggregate/aggregate_request.c

```diff
@@ -1232,13 +1226,12 @@ int buildOutputPipeline(AREQ *req, QueryError *status) {
     RLookup *lookup = AGPLN_GetLookup(pln, NULL, AGPLN_GETLOOKUP_LAST);
     for (size_t ii = 0; ii < req->outFields.numFields; ++ii) {
       ReturnedField *ff = req->outFields.fields + ii;
-      RLookupKey *kk = RLookup_GetKey_TEMP(lookup, ff->name, RLOOKUP_F_NOFLAGS);
+      RLookupKey *kk = RLookup_GetKey(lookup, ff->name, RLOOKUP_M_READ, RLOOKUP_F_NOFLAGS);
       if (!kk) {
         QueryError_SetErrorFmt(status, QUERY_ENOPROPKEY, "No such property `%s`", ff->name);
         goto error;
-      } else if (!(kk->flags & (RLOOKUP_F_SCHEMASRC | RLOOKUP_F_SVSRC))) {
-        // TODO: this is a dead code
-        QueryError_SetErrorFmt(status, QUERY_EINVAL, "Property `%s` is not in document", ff->name);
+      } else if (!(kk->flags & RLOOKUP_F_SCHEMASRC)) {
+        QueryError_SetErrorFmt(status, QUERY_EINVAL, "Property `%s` is not known to schema", ff->name);
```

Comment:
Consider rephrasing "not known to schema" according to existing messages, to use `QUERY_ENOPROPKEY`
or
"not specified in schema" or "not in schema"
https://github.com/RediSearch/RediSearch/blob/ed988bdb5a40dbbd296a6b5a57082bdd3a627734/src/document.c#L109
https://github.com/RediSearch/RediSearch/blob/ed988bdb5a40dbbd296a6b5a57082bdd3a627734/src/aggregate/aggregate_request.c#L1080

And see if you can get it covered in a test